Add support for serial protocol, plus factory reset command

tve opened the pull request:
I needed serial support and added it. Example 11 shows its use as well as code to change baud rate (which can be a bit tricky).

nseidle commented:
This looks great! Thank you. I'll give it a try tonight.

nseidle commented:
I'm impressed you dug in deep enough into my code to figure out the packetCfg structure. I like factoryReset().

I really want to accept this PR but there's some issues.

* Serial.printf() doesn't compile. Please assume user has stock Arduino without [printf wrapper](https://playground.arduino.cc/main/printf).
* Use debug.print instead of Serial.print so that user can pipe debug to port of their choice.
* (Nit) [Use a uint8_t](https://github.com/sparkfun/SparkFun_Ublox_Arduino_Library/pull/2/files?w=1#diff-09b68aa0742e5325ae8fb037d77970ebR71) instead of an int, save a byte.
* Split [hard reset](https://github.com/sparkfun/SparkFun_Ublox_Arduino_Library/pull/2/files?w=1#diff-09b68aa0742e5325ae8fb037d77970ebR80) out to its own function. I use hard reset (cold start) to test TTFF more than I use factory reset.
* Some modules (ZED-F9P) have two UARTs. If would be nice to have setBaudRate() support multiple ports.

Overall, I like it a lot. Thanks for the help! I've got a SAM-M8Q working (surprisingly indoors) using serial. Works well.

tve commented:
I believe I addressed all your concerns. I cannot test setSerialRate on the second uart port as my devices don't have one. The code for this is simple, though. LMK if there's more you'd like see changed.

nseidle commented:
Excellent, excellent. All my concerns are addressed. Thank you very much.

nseidle commented:
You additions should be available in v1.2.0 soon. I modified Ex11 to use software serial as most users (made up statistic) don't have Serial1.
